classes/ActionScheduler_WPCLI_QueueRunner.php

```diff
@@ -0,0 +1,220 @@
+	 *
+	 * @return int The number of actions that will be run.
+	 */
+	public function setup( $batch_size, $force = false ) {
+		$this->run_cleanup();
+		$this->add_hooks();
+
+		// Check to make sure there aren't too many concurrent processes running.
+		$claim_count = $this->store->get_claim_count();
+		$too_many    = $claim_count >= apply_filters( 'action_scheduler_queue_runner_concurrent_batches', 5 );
+		if ( $too_many ) {
+			if ( $force ) {
+				WP_CLI::warning( __( 'There are too many concurrent batches, but the run is forced to continue.', 'action-scheduler' ) );
+			} else {
+				WP_CLI::error( __( 'There are too many concurrent batches.', 'action-scheduler' ) );
+			}
+		}
+
+		// Stake a claim and store it.
+		$this->claim = $this->store->stake_claim( $batch_size );
+		$this->monitor->attach( $this->claim );
+		$this->actions = $this->claim->get_actions();
+		$this->id      = $this->claim->get_id();
+
+		return count( $this->actions );
+	}
+
+	/**
+	 * Run the queue cleaner.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function run_cleanup() {
+		$this->cleaner->clean();
+	}
+
+	/**
+	 * Add our hooks to the appropriate actions.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function add_hooks() {
+		add_action( 'action_scheduler_before_execute', array( $this, 'before_execute' ) );
+		add_action( 'action_scheduler_after_execute', array( $this, 'after_execute' ) );
+		add_action( 'action_scheduler_failed_execution', array( $this, 'action_failed' ) );
+	}
+
+	/**
+	 * Set up the WP CLI progress bar.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function setup_progress_bar() {
+		$count              = count( $this->actions );
+		$this->progress_bar = \WP_CLI\Utils\make_progress_bar(
+			sprintf( _n( 'Running %d task', 'Running %d tasks', $count, 'action-scheduler' ), number_format_i18n( $count ) ),
+			$count
+		);
+	}
+
+	/**
+	 * Ensure the progress bar has finished properly.
+	 *
+	 * @author Jeremy Pry
+	 */
+	protected function finish_progress_bar() {
+		$this->progress_bar->finish();
+	}
+
+	/**
+	 * Process actions in the queue.
+	 *
+	 * @author Jeremy Pry
+	 * @return int The number of actions processed.
+	 */
+	public function run() {
+		$this->setup_progress_bar();
+		foreach ( $this->actions as $action_id ) {
+			// Error if we lost the claim.
+			$all_actions = array_flip( $this->store->find_actions_by_claim_id( $this->id ) );
+			if ( ! array_key_exists( $action_id, $all_actions ) ) {
+				$this->finish_progress_bar();
+				WP_CLI::error( __( 'The claim has been lost. Aborting.', 'action-scheduler' ) );
+			}
+
+			$this->process_action( $action_id );
+			$this->progress_bar->tick();
+
+			// Free up memory after every 50 items
+			if ( 0 === $this->progress_bar->current() % 50 ) {
+				$this->stop_the_insanity();
+			}
+		}
+
+		$completed = $this->progress_bar->current();
+		$this->finish_progress_bar();
+
+		return $completed;
+	}
```

Comment:
Because this instance of `run()` isn't calling `'action_scheduler_before_process_queue'` and `'action_scheduler_after_process_queue'`, comment counting isn't being deferred by `ActionScheduler_wpCommentLogger`. Not sure if that makes much/any perf difference in the WP CLI context and not a major issue, but I thought it was something worth noting.